core/src/main/java/com/linecorp/armeria/client/retry/RetryRule.java

```diff
@@ -382,10 +385,10 @@ default RetryRule orElse(RetryRule other) {
     CompletionStage<RetryDecision> shouldRetry(ClientRequestContext ctx, @Nullable Throwable cause);
 
     /**
-     * Returns whether this rule requires the response trailers to determine if a {@link Response} is
-     * successful or not.
+     * Returns a set of {@link RequestLogProperty} to be required to evaluate this rule and to determine
+     * if a {@link Response} is successful or not.
      */
-    default boolean requiresResponseTrailers() {
-        return false;
+    default Set<RequestLogProperty> getRequiredLogProperties() {
+        return Collections.emptySet();
```

Comment:
Should we revert this? `RetryRule` is an API created for users to inherit and implement.
The breaking change would not be acceptable to them in a minor upgrade.

Reply:
Updated!